src/compiler/factory.ts

```diff
@@ -1982,7 +1982,7 @@ namespace ts {
         node.decorators = asNodeArray(decorators);
         node.modifiers = asNodeArray(modifiers);
         node.isExportEquals = isExportEquals;
-        node.expression = expression;
+        node.expression = parenthesizeBinaryOperand(SyntaxKind.EqualsToken, expression, /*isLeftSideOfBinary*/ false, /*leftOperand*/ undefined);

```

Comment:
As far as I can tell, `export default <expr>` binds as tightly as `export = <expr>`, so this seems appropriate.

Reply:
Actually, I think `export default` may need its own parenthesization logic. [Per the spec](https://tc39.github.io/ecma262/#prod-ExportDeclaration), `export default` accepts _AssigmentExpression_ but has a lookahead restriction for `function`, `async function`, and `class`. 

Basically, that means you need to parenthesize in the following cases:

- BinaryExpression of CommaToken
- CommaList (synthetic list of multiple comma expressions)
- FunctionExpression
- ClassExpression

Reply:
Actually, I think `export default` may need its own parenthesization logic. [Per the spec](https://tc39.github.io/ecma262/#prod-ExportDeclaration), `export default` accepts _AssigmentExpression_ but has a lookahead restriction for `function`, `async function`, and `class`. 

Basically, that means you need to parenthesize in the following cases:

- BinaryExpression of CommaToken
- CommaList (synthetic list of multiple comma expressions)
- FunctionExpression
- ClassExpression

Reply:
done.